src/cloudflare/internal/vectorize-api.ts

```diff
@@ -45,14 +45,20 @@ class VectorizeIndexImpl implements Vectorize {
     options?: VectorizeQueryOptions
   ): Promise<VectorizeMatches> {
     if (this.indexVersion === 'v2') {
-      if (
-        options &&
-        options.returnMetadata &&
-        !isVectorizeMetadataRetrievalLevel(options.returnMetadata)
-      ) {
-        throw new Error(
-          `Invalid returnMetadata option. Expected: "none", "indexed" or "all"; got: ${options.returnMetadata}`
-        );
+      if (options && options.returnMetadata) {
+        if (
+          typeof options.returnMetadata !== 'boolean' &&
```

Comment:
Maybe perform this type check once?

Reply:
How would you recommend this to be refactored?